server/drivers/xosdlib_drv.c

```diff
@@ -85,74 +80,53 @@ xosdlib_drv_init (Driver *drvthis)
-					drvthis->name, size, DEFAULT_SIZE);
-			sscanf(DEFAULT_SIZE, "%dx%d", &w, &h);
-		}
-		p->width = w;
-		p->height = h;
-	}
-	else {
-		/* Determine the size of the screen */
+	if ((drvthis->request_display_width() > 0)
+	    && (drvthis->request_display_height() > 0)) {
+		/* If this driver is secondary driver, use size from primary driver */
 		p->width = drvthis->request_display_width();
 		p->height = drvthis->request_display_height();
-		if ((p->width <= 0) || (p->width >= LCD_MAX_WIDTH) ||
-		    (p->height <= 0) || (p->height >= LCD_MAX_HEIGHT)) {
-			p->width = LCD_DEFAULT_WIDTH;
-			p->height = LCD_DEFAULT_HEIGHT;
+	}
+	else {
+		/* Use our own size from config file */
+		if ((sscanf(config.size , "%dx%d", &p->width, &p->height) != 2)
+		    || (p->width <= 0) || (p->width > LCD_MAX_WIDTH)
+		    || (p->height <= 0) || (p->height > LCD_MAX_HEIGHT)) {
+			report(RPT_WARNING, "%s: cannot read Size: %s; using default %s",
+					drvthis->name, config.size, DEFAULT_SIZE);
+			sscanf(DEFAULT_SIZE, "%dx%d", &p->width, &p->height);
 		}
 	}
 	report(RPT_INFO, "%s: using size %dx%d", drvthis->name, p->width, p->height);
 
 	/* Which x/y offsets */
-	offset = drvthis->config_get_string(drvthis->name, "Offset", 0, DEFAULT_OFFSET);
-	debug(RPT_INFO, "%s: Offset (in config) is '%s'", __FUNCTION__, offset);
-	if (sscanf(offset, "%dx%d", &x, &y) != 2) {
+	debug(RPT_INFO, "%s: Offset (in config) is '%s'", __FUNCTION__, config.offset);
+	if (sscanf(config.offset, "%dx%d", &p->xoffs, &p->yoffs) != 2) {
 		report(RPT_WARNING, "%s: cannot read Offset: %s. using default %s",
-				drvthis->name, offset, DEFAULT_OFFSET);
-		sscanf(DEFAULT_OFFSET, "%dx%d", &x, &y);
+				drvthis->name, config.offset, DEFAULT_OFFSET);
+		sscanf(DEFAULT_OFFSET, "%dx%d", &p->xoffs, &p->yoffs);
 	}
-	p->xoffs= x;
-	p->yoffs = y;
-
 	report(RPT_INFO, "%s: using offset %dx%d", drvthis->name, p->xoffs, p->yoffs);
 
 
 	/* Which backlight brightness */
-	tmp = drvthis->config_get_int(drvthis->name, "Brightness", 0, DEFAULT_BRIGHTNESS);
-	debug(RPT_INFO, "%s: Brightness (in config) is '%d'", __FUNCTION__, tmp);
-	if ((tmp < 0) || (tmp > 1000)) {
-		report(RPT_WARNING, "%s: Brightness must be between 0 and 1000; using default %d",
-			drvthis->name, DEFAULT_BRIGHTNESS);
-		tmp = DEFAULT_BRIGHTNESS;
-	}
-	p->brightness = tmp;
+	debug(RPT_INFO, "%s: Brightness (in config) is '%d'", __FUNCTION__, config.brightness);
+	p->brightness = config.brightness;
 
 	/* Which backlight-off "brightness" */
-	tmp = drvthis->config_get_int(drvthis->name, "OffBrightness", 0, DEFAULT_OFFBRIGHTNESS);
-	debug(RPT_INFO, "%s: OffBrightness (in config) is '%d'", __FUNCTION__, tmp);
-	if ((tmp < 0) || (tmp > 1000)) {
-		report(RPT_WARNING, "%s: OffBrightness must be between 0 and 1000; using default %d",
-			drvthis->name, DEFAULT_OFFBRIGHTNESS);
-		tmp = DEFAULT_OFFBRIGHTNESS;
-	}
-	p->offbrightness = tmp;
+	debug(RPT_INFO, "%s: OffBrightness (in config) is '%d'", __FUNCTION__, config.offbrightness);
```

Comment:
Are debug statements like these everywhere?

Reply:
The whole LCDproc code base is very inconsistent (with formatting, logging, etc). The recommendation seems to be to just keep going in the style of the original author. 

I neither added nor deleted debug or report statements AFAIK.

Reply:
> The recommendation seems to be to just keep going in the style of the original author.

Yes, that makes sense. It is too much effort to make this consistent now.